Sources/Sentry/Profiling/SentryContinuousProfiler.mm

```diff
@@ -36,19 +42,24 @@
 _sentry_threadUnsafe_transmitChunkEnvelope(void)
 {
     const auto profiler = _threadUnsafe_gContinuousCurrentProfiler;
-    const auto stateCopy = [profiler.state copyProfilingData];
-    const auto startSystemTime = profiler.continuousChunkStartSystemTime;
-    const auto endSystemTime = SentryDependencyContainer.sharedInstance.dateProvider.systemTime;
-    profiler.continuousChunkStartSystemTime = endSystemTime + 1;
+    const auto profilerState = [profiler.state copyProfilingData];
     [profiler.state clear]; // !!!: profile this to see if it takes longer than one sample duration
                             // length: ~9ms
 
+    const auto metricProfilerState = [profiler.metricProfiler serializeContinuousProfileMetrics];
+    [profiler.metricProfiler clear];
+
+#    if SENTRY_HAS_UIKIT
+    const auto framesTracker = SentryDependencyContainer.sharedInstance.framesTracker;
+    SentryScreenFrames *screenFrameData = [framesTracker.currentFrames copy];
```

Comment:
So this was actually copypasta'd from the trace profiler's usage pattern, where we need to keep a copy in case the entire structure is mutated by another profiler, to avoid either data loss or corruption.

Since the frames tracker starts/stops with the profiler, we probably don't have to do this anymore.

Reply:
I think what happened is when I originally read the callsite in the past, `currentFrames` looks like a property that is maintaining state, instead of creating an instance of something to return, so I took a copy to avoid data loss/races/corruption. This is another casualty in the long battle over whether to use dot or bracket notation when calling an objc method. Also, it used to be declared as a `@property`, which I've changed in this PR. Never use `@property` synthesis to make a getter that does something unrelated–always use a method for that!